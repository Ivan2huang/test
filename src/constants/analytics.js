export const CATEGORIES = {
  HOME_SCREEN: 'Home Screen',
  CLAIMS_PAGE: 'Claims',
  CLAIMS_FILTER: 'Claims filter',
  CLAIMS_SUBMISSION: 'Claims submission',
  EHEALTH_CARD_PAGE: 'Profile, e-health card',
  PANEL_SEARCH_RESULT: 'Panel Search results',
  PANEL_CLINIC_SEARCH: 'Panel clinic search',
  PANEL_CLINIC_PAGE: 'Panel clinics',
  PANEL_CLINIC_DETAIL: 'Panel clinic details',
  BENEFITS_PAGE: 'Profile, benefits summary',
  DETAILS_PAGE: 'Profile, my details',
  DOCUMENTS_PAGE: 'Profile, Forms and Documents',
  HELP_PAGE: 'Profile, help',
  SETTINGS_PAGE: 'Profile, settings',
  MY_LIFESTYLE_OVERVIEW: 'My lifestyle overview',
  HEALTH_QUESTIONNAIRE: 'Health Questionnaire',
  NAVIGATION_BAR: 'Navigation bar',
};

export const LABEL = {
  USER_TYPE: 'user_type',
};

export const EVENT_TYPES = {
  USER_ACTION: 'user_action',
};
