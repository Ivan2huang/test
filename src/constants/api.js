/* eslint-disable import/prefer-default-export */

export const SERVICE_IDENTIFIERS = {
  authService: '/auth-service/api/v1',
  authServiceV2: '/auth-service/api/v2/clients',
  claimService: '/claim-service/api/v1/clients',
  memberService: '/member-service/api/v1/clients',
  benefitService: '/benefit-service/api/v1/clients',
  walletService: '/wallet-service/api/v1/clients',
  eWalletService: '/e-wallet-service/api/v1/clients',
  contentService: '/content-service/api/v1',
  wellnessService: '/wellness-service/api/v1/clients',
  recommendationService: '/recommendation-service/api/v1/clients',
  paymentService: '/payment-service/api/v1/clients',
  zuulService: '/zuul-service',
};

export const FUNC_IDENTIFIERS = {
  funcContentProduct: '/func-content-product/api/v1',
  funcContentContacts: '/func-content-contacts/api/v1',
  funcContentDocuments: '/func-content-documents/api/v1',
  funcContentHelp: '/func-content-help/api/v1',
  funcContentTermsConditions: '/func-content-terms-conditions/api/v1',
  funcContentPrivacyPolicy: '/func-content-privacy-policy/api/v1',
  funcContentWellness: '/func-content-wellness/api/v1',
  funcExternalBenefits: '/func-benefit-handle-external-benefits/api/v1',
  funcBenefits: '/func-benefit-handle-benefits/api/v1',
  funcPanels: '/func-panel-handle-panels/api/v1',
  funcContentDownloadDocument: '/func-content-documents-download/api/v1',
  funcContentNote: '/func-content-note/api/v1',
  funcContentTip: '/func-content-tip/api/v1',
};
