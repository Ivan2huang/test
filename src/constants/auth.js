import paths from '../helpers/paths';

export const ID_TOKEN = 'id_token';
export const ACCESS_TOKEN = 'access_token';
export const REFRESH_TOKEN = 'refresh_token';
export const LOGGED_IN = 'logged_in';
export const USER_ID = 'user_id';
export const CLIENT_ID = 'client_id';
export const AUTH_CLIENT_ID = 'authClientId';
export const PENDING_ACTIONS = 'pending_actions';
export const PAYMENT_JWT = 'payment_jwt';
export const LANG = 'lang';
export const CUSTOMTOKEN_HEADER = 'cxa-authorization';
export const error = {
  login: {
    messageKeys: {
      Unauthorized: 'login.error.unauthorized',
    },
  },
  token: {
    invalidRefreshToken: 'refresh_token_invalid',
  },
};

export const routesWithoutFetchData = [
  paths.common.accountAcivation,
  paths.common.accountActivationError,
  paths.common.error,
];
