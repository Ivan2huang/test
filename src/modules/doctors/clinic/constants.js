const CLINIC_LIST_VIEW_TYPE = {
  selectedStackedClinics: 'selectedStackedClinics',
  filteredClinics: 'filteredClinics',
};

const FILTER_BY = {
  consultationType: 'consultationType',
};

const VIEW = {
  list: 'list',
  map: 'map',
};

export { CLINIC_LIST_VIEW_TYPE, FILTER_BY, VIEW };
