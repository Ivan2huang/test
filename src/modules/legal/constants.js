// eslint-disable-next-line import/prefer-default-export
export const LOCALE_STORAGE = 'LOCALE_STORAGE';
