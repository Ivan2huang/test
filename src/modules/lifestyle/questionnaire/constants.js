export const QUESTIONNAIRE_TYPES = {
  aboutMe: 'aboutMe',
  choices: 'choices',
  health: 'health',
  futureMe: 'futureMe',
};

export const BMI_UNIT = {
  weight: {
    KG: 'Kg',
    LB: 'Lb',
  },
  height: {
    CM: 'Cm',
    IN: 'In',
    FTIN: 'FtIn',
  },
};

export const UPLOAD_RULE_ID = 'upload-rule';

export const QUESTIONNAIRE_FORM = 'lifestyle-questionnaire';
