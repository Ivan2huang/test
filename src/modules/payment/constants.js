const ADD_NEW_CARD = 'ADD_NEW_CARD';

const PAYMENT_FORM = {
  cardNumber: {
    gridColumns: { xs: 12, md: 12 },
    defaultPlaceholder: 'Card Number',
  },
  expiryMonth: {
    gridColumns: { xs: 6, md: 6 },
    defaultPlaceholder: 'Expiry Month',
  },
  expiryYear: {
    gridColumns: { xs: 6, md: 6 },
    defaultPlaceholder: 'Expiry Year',
  },
  securityCode: {
    gridColumns: { xs: 12, md: 12 },
    defaultPlaceholder: 'Security Code',
    inputType: 'password',
  },
};

const CREDIT_CARD_NUMBER_LENGTH = 16;

const CREDIT_CARD_NUMBER_MASK_CHAR = '*';

const REQUIRED_3DS = 'REQUIRED_3DS';

const ERROR = 'ERROR';

const UNKNOWN = 'UNKNOWN';

const ECONNABORTED = 'ECONNABORTED';

const BACK = 'BACK';

const OPEN_TERM_OF_USE_MESSAGE_TYPE = 'OPEN_TERM_OF_USE';

const CALLBACK_METHOD = {
  GET: 'GET',
  POST: 'POST',
};

const PAYMENT_METHOD_CC_SEPARATOR = '_';

const PAYMENT_CARD_FORM_NAME = 'card';

export {
  ADD_NEW_CARD,
  PAYMENT_FORM,
  CREDIT_CARD_NUMBER_LENGTH,
  CREDIT_CARD_NUMBER_MASK_CHAR,
  REQUIRED_3DS,
  ERROR,
  UNKNOWN,
  ECONNABORTED,
  BACK,
  OPEN_TERM_OF_USE_MESSAGE_TYPE,
  CALLBACK_METHOD,
  PAYMENT_METHOD_CC_SEPARATOR,
  PAYMENT_CARD_FORM_NAME,
};
