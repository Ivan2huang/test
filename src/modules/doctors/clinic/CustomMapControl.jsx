import React, { useEffect } from 'react';
import { createPortal } from 'react-dom';
import { MAP } from 'react-google-maps/lib/constants';
import PropTypes from 'prop-types';

import Box from '@material-ui/core/Box';

const CustomMapControl = ({ position, children }, context) => {
  const map = context[MAP];

  const controlDiv = document.createElement('div');

  useEffect(() => {
    const controls = map.controls[position];
    controls.push(controlDiv);
  });
  return createPortal(<Box>{children}</Box>, controlDiv);
};

CustomMapControl.contextTypes = {
  [MAP]: PropTypes.shape({}),
};

CustomMapControl.propTypes = {
  position: PropTypes.number.isRequired,
  children: PropTypes.element.isRequired,
  context: PropTypes.shape({}),
};

export default CustomMapControl;
