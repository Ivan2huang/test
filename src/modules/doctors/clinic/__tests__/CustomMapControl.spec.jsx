import React from 'react';
import { render } from '@testing-library/react';
import { MAP } from 'react-google-maps/lib/constants';

import PropTypes from 'prop-types';
import CustomMapControl from '../CustomMapControl';

describe('CustomMapControl component', () => {
  it('should render the external component', () => {
    class Component extends React.Component {
      getChildContext() {
        return {
          [MAP]: {
            controls: [[]],
          },
        };
      }

      render() {
        // eslint-disable-next-line react/prop-types,react/destructuring-assignment
        return <div>{this.props.children}</div>;
      }
    }
    Component.childContextTypes = {
      [MAP]: PropTypes.shape({}),
    };
    const { container } = render(
      <Component>
        <CustomMapControl position={0}>
          <div>Dummy</div>
        </CustomMapControl>
      </Component>,
    );

    expect(container).toMatchSnapshot();
  });
});
