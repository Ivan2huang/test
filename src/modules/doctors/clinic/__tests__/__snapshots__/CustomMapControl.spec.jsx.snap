// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMapControl component should render the external component 1`] = `
<div>
  <div />
</div>
`;
